src/promptflow-tools/promptflow/tools/common.py

```diff
@@ -120,9 +135,13 @@ def parse_chat(chat_str, valid_roles: List[str] = None):
                                 "or view sample 'How to use functions with chat models' in our gallery.")
                 # "name" is optional for other role types.
                 else:
-                    last_message["content"] = chunk
+                    last_message["content"] = to_content_str_or_list(chunk, hash2images)
             else:
-                last_message["name"], last_message["content"] = parsed_result
+                if last_message["role"] == "function":
```

Comment:
Why do we need to distinct function role here? Please add some comments.

Reply:
not necessary